### PR TITLE
basic login functionality

### PR DESCRIPTION
**Pull request by mingyokim**
Basic user login page and login confirmation page.

Used https://github.com/firebase/firebaseui-web-react.

You can checkout and test locally with your own gmail account or test@example.com with pw=testtest.

This PR is built on top of #3, which initializes the firebase module.

### CODE REVIEW

File: web/components/login/SignedIn.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export default class SignedIn extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: null,
+    };
+  }
+
+  componentWillMount() {
+    this.removeAuthListener = firebase.auth().onAuthStateChanged((user) => {
```
Comment (bobheadxi) [inline web/components/login/SignedIn.js]: did we decide on promises vs async-await?

File: web/components/login/SignedIn.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      }
+    });
+  }
+
+  render() {
+    let message = '';
+
+    if (this.state.user) {
+      // User is signed in.
+      message = `Signed in as ${this.state.user.displayName}`;
```
Comment (bobheadxi) [inline web/components/login/SignedIn.js]: 🔥

File: web/package.json [hunk 3]
```diff
@@ -31,6 +31,8 @@
     "prop-types": "15.6.1",
     "react": "16.4.0",
     "react-dom": "16.4.0",
+    "react-firebaseui": "^3.0.4",
```
Comment (bobheadxi) [inline web/package.json]: Using `^` means that every time we install dependencies, the manager will grab the newest version - while semver ideally enforces that no breaking changes will occur, it's probably best to lock it down by removing `^` and only upgrading explicitly. This helps avoid `lock`file diffs polluting commit history.

**Comment by bobheadxi** [comment]
Also reviewed the changes in the other PR because i didnt get a chance to - left some things that you might want to take note of: https://github.com/nwhacks/nwhacks2019/pull/3#pullrequestreview-131886969

File: web/components/login/SignedIn.js (hunk 1 above)
Comment (mingyokim) [reply web/components/login/SignedIn.js]: nope

File: web/package.json (hunk 3 above)
Comment (mingyokim) [reply web/package.json]: do we want to do that for every package in this file?

**Pull request merged**